### ISSUE TITLE
before handler soft delete error

### ISSUE THREAD

**Issue by VyseExhale** [mevdschee/php-crud-api#209]
Hi there,

yesterdays updated raised an error working with softdeletes like the example: 
```php
'before'=>function(&$cmd, &$db, &$tab, &$id, &$in) { 
	if ($cmd == 'delete') {
		$cmd = 'update'; // change command to update
		foreach($in as $k => $o) {
			$in[$k]->deleted = date('Y-m-d H:i:s', time());
		}				
	}
			
},
'column_authorizer'=>function($cmd, $db ,$tab, $col) { 
	return ( ! in_array($col, array('deleted')));
},
'record_filter'=>function($cmd,$db,$tab) { 
	return array('deleted,is,null');
}
```

in the before we change the $cmd delete to update... updating the "deleted" column to a current date value.... in the column_authorizer we filter out the "deleted" column, so we don't show it in our results

But because the [applyBeforeHandler](https://github.com/mevdschee/php-crud-api/blob/master/api.php#L1868-L1870) is called before the [applyColumnAuthorizer](https://github.com/mevdschee/php-crud-api/blob/master/api.php#L1881)
the "deleted" column gets filtered out... Now the row column deleted will never be set to the current_date.. and so it will never be softdeleted neither will it be harddeleted.

A solution would be is to call the applyBeforeHandler after the applyColumnAuthorizer..

**Comment by mevdschee** [comment]
Fixed in 753c0c05bb21a4f9d1fb84c967e2646161251d80

**Comment by VyseExhale** [comment]
Wow you're superfast
